Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp

```diff
@@ -2509,6 +2549,64 @@ auto B3IRGenerator::addArrayLen(ExpressionType arrayref, ExpressionType& result)
     return { };
 }
 
+auto B3IRGenerator::addStructNew(uint32_t typeIndex, Vector<ExpressionType>& args, ExpressionType& result) -> PartialResult
+{
+    const auto type = Type { TypeKind::Ref, m_info.typeSignatures[typeIndex]->index() };
+
+    // FIXME: inline the allocation.
+    // https://bugs.webkit.org/show_bug.cgi?id=244388
+    Value* structValue = m_currentBlock->appendNew<CCallValue>(m_proc, toB3Type(type), origin(),
+        m_currentBlock->appendNew<ConstPtrValue>(m_proc, origin(), tagCFunction<OperationPtrTag>(operationWasmStructNewEmpty)),
+        instanceValue(),
+        m_currentBlock->appendNew<Const32Value>(m_proc, origin(), typeIndex));
+
+    const auto& structType = *m_info.typeSignatures[typeIndex]->template as<StructType>();
+    for (uint32_t i = 0; i < args.size(); ++i)
+        emitStructSet(structValue, i, structType, args[i]);
+
+    result = push(structValue);
+
+    return { };
+}
+
+auto B3IRGenerator::addStructGet(ExpressionType structReference, const StructType& structType, uint32_t fieldIndex, ExpressionType& result) -> PartialResult
+{
+    Value* structValue = get(structReference);
+    int32_t payloadFieldOffset = fixupPointerPlusOffset(structValue, JSWebAssemblyStruct::offsetOfPayload());
```

Comment:
We do not need this. `offsetOfPayload()` is constant and small. Rather, considering about the possibility of rewritten value of structValue makes the code hard to read.